src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,13 +753,15 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
     private final int[] decodeInts(final String string) {
-        final int nValues = ParsingUtils.split(string, INT_DECODE_ARRAY, ',');
+        // find number of CSV items to set array size
+        int numItems = StringUtils.countMatches(string, ",") + 1;
+        String[] intDecodeArray = new String[numItems];
+        final int nValues = ParsingUtils.split(string, intDecodeArray, ',');
```

Comment:
If I understand, the reason for avoiding `String.split` was poor performance that was fixed in JDK7. Since htsjdk support for JDK6 is planned to be dropped in 5 months, is it worth it to add an additional dependency now that may be moot in that time?

(It would take some benchmarking to confirm my suspicions).